torch/_dynamo/variables/builder.py

```diff
@@ -440,6 +440,31 @@ def index_source(key):
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
         # NB: These can't be put in type_dispatch, they have to run later
+        elif value in {
```

Comment:
You also have to do this before the `is_allowed` check or allowlist shuffling will short circuit past this.

Reply:
this part is going over my head.

How come it works for my unit test and not real fsdp?